upip/upip_utarfile.py

```diff
@@ -39,7 +39,13 @@ def readinto(self, buf):
         return sz
 
     def skip(self):
-        self.f.read(self.content_len + self.align)
+        sz = self.content_len + self.align
+        if sz:
+            buf = bytearray(16)
+            while sz:
+                s = min(sz, 16)
+                self.f.readinto(buf, s)
+                sz -= s
```

Comment:
My first motion was to implement seek(SET_CUR). But then I looked that no .ioctl was implemented in uzlib.DecompIO at all, plus it's not possible to provide correct return value for it. And I thought: why bother for a single usecase? If you think that implementing that (with invalid return value, like -1) is useful, I can queue it up.